File: Proyectos/TI.1 PAO202650_SINBDD_NRC30742/TI.1.3. PAO202650_RESTFUL_JAVA_SINBDD_NRC30742/04. SERVIDOR/CONUNI_RESTFUL_JAVA_GR05/src/main/java/ec/edu/monster/controlador/AuthController.java
package ec.edu.monster.controlador;

import jakarta.ws.rs.Consumes;
import jakarta.ws.rs.POST;
import jakarta.ws.rs.Path;
import jakarta.ws.rs.Produces;
import jakarta.ws.rs.core.MediaType;
import jakarta.ws.rs.core.Response;

/**
 * Controlador REST para la autenticación de usuarios.
 * <p>
 * Las credenciales residen ÚNICAMENTE en el servidor. Los clientes
 * envían usuario y contraseña vía POST y el servidor responde si son válidas.
 * </p>
 *
 * <p>Endpoint: {@code POST /api/auth/login}</p>
 * <p>Body JSON: {@code { "usuario": "...", "contrasena": "..." }}</p>
 * <p>Respuesta: {@code { "autenticado": true/false, "mensaje": "..." }}</p>
 *
 * @author GR09 - Arquitectura de Software
 */
@Path("/auth")
@Produces(MediaType.APPLICATION_JSON)
@Consumes(MediaType.APPLICATION_JSON)
public class AuthController {

    // Credenciales almacenadas SOLO en el servidor
    private static final String USUARIO_VALIDO    = "MONSTER";
    private static final String CONTRASENA_VALIDA = "MONSTER9";

    /**
     * Endpoint de autenticación.
     * Recibe credenciales del cliente y retorna si son válidas.
     *
     * @param credenciales objeto JSON con "usuario" y "contrasena"
     * @return 200 OK con {@code { "autenticado": true }}  o
     *         401 Unauthorized con {@code { "autenticado": false, "mensaje": "Credenciales incorrectas" }}
     */
    @POST
    @Path("/login")
    public Response login(LoginRequest credenciales) {
        if (credenciales == null
                || credenciales.getUsuario() == null
                || credenciales.getContrasena() == null) {
            return Response.status(Response.Status.BAD_REQUEST)
                    .entity(new LoginResponse(false, "Credenciales no proporcionadas"))
                    .build();
        }

        boolean esValido = USUARIO_VALIDO.equals(credenciales.getUsuario().trim().toUpperCase())
                        && CONTRASENA_VALIDA.equals(credenciales.getContrasena().trim());

        System.out.println("DEBUG: login – usuario='" + credenciales.getUsuario()
                + "' autenticado=" + esValido);

        if (esValido) {
            return Response.ok(new LoginResponse(true, "Autenticación exitosa")).build();
        } else {
            return Response.status(Response.Status.UNAUTHORIZED)
                    .entity(new LoginResponse(false, "Credenciales incorrectas"))
                    .build();
        }
    }

    // ====== DTOs internos al controlador ======

    /**
     * DTO para recibir las credenciales del cliente.
     */
    public static class LoginRequest {
        private String usuario;
        private String contrasena;

        public LoginRequest() {}

        public String getUsuario()    { return usuario; }
        public void setUsuario(String u) { this.usuario = u; }

        public String getContrasena()    { return contrasena; }
        public void setContrasena(String c) { this.contrasena = c; }
    }

    /**
     * DTO para devolver el resultado de la autenticación.
     */
    public static class LoginResponse {
        private boolean autenticado;
        private String  mensaje;

        public LoginResponse() {}
        public LoginResponse(boolean autenticado, String mensaje) {
            this.autenticado = autenticado;
            this.mensaje     = mensaje;
        }

        public boolean isAutenticado()   { return autenticado; }
        public void setAutenticado(boolean a) { this.autenticado = a; }

        public String getMensaje()   { return mensaje; }
        public void setMensaje(String m) { this.mensaje = m; }
    }
}

File: Proyectos/TI.1 PAO202650_SINBDD_NRC30742/TI.1.4. PAO202650_RESTFUL_DOTNET_SINBDD_NRC30742/04. SERVIDOR/CONUNI_RESTFUL_DOTNET_GR5/CONUNI_RESTFUL_DOTNET_GR9S/Controllers/AuthController.cs
using System;
using System.Net;
using System.Web.Http;

namespace CONUNI_RESTFUL_DOTNET_GR9S.Controllers
{
    /// <summary>
    /// Controlador REST para la autenticación de usuarios.
    /// Las credenciales residen ÚNICAMENTE en el servidor.
    /// Los clientes envían usuario y contraseña vía POST y el servidor
    /// responde si son válidas.
    /// <para>Endpoint: POST api/auth/login</para>
    /// </summary>
    [RoutePrefix("api/auth")]
    public class AuthController : ApiController
    {
        // Credenciales almacenadas SOLO en el servidor
        private const string UsuarioValido    = "MONSTER";
        private const string ContrasenaValida = "MONSTER9";

        /// <summary>
        /// Valida las credenciales del usuario.
        /// </summary>
        /// <param name="credenciales">Objeto con usuario y contraseña</param>
        /// <returns>200 OK con autenticado=true, o 401 Unauthorized con autenticado=false</returns>
        [HttpPost]
        [Route("login")]
        public IHttpActionResult Login([FromBody] LoginRequest credenciales)
        {
            if (credenciales == null
                || string.IsNullOrEmpty(credenciales.Usuario)
                || string.IsNullOrEmpty(credenciales.Contrasena))
            {
                return BadRequest("Credenciales no proporcionadas");
            }

            bool esValido = UsuarioValido.Equals(credenciales.Usuario.Trim().ToUpper())
                         && ContrasenaValida.Equals(credenciales.Contrasena.Trim());

            System.Diagnostics.Debug.WriteLine(
                $"DEBUG: Login – usuario='{credenciales.Usuario}' autenticado={esValido}");

            if (esValido)
            {
                return Ok(new LoginResponse { Autenticado = true, Mensaje = "Autenticación exitosa" });
            }
            else
            {
                return Content(HttpStatusCode.Unauthorized,
                    new LoginResponse { Autenticado = false, Mensaje = "Credenciales incorrectas" });
            }
        }
    }

    /// <summary>DTO para recibir las credenciales del cliente.</summary>
    public class LoginRequest
    {
        public string Usuario    { get; set; }
        public string Contrasena { get; set; }
    }

    /// <summary>DTO para devolver el resultado de la autenticación.</summary>
    public class LoginResponse
    {
        public bool   Autenticado { get; set; }
        public string Mensaje     { get; set; }
    }
}
